bugsnag-plugin-react-native/src/main/java/com/bugsnag/android/ThreadDeserializer.java

```diff
@@ -32,6 +32,7 @@ public Thread deserialize(Map<String, Object> map) {
                 MapUtils.<String>getOrThrow(map, "name"),
                 ThreadType.valueOf(type.toUpperCase(Locale.US)),
                 errorReportingThread,
+                Thread.State.byName(MapUtils.<String>getOrThrow(map, "state")),
```

Comment:
Will this require any updates to the React Native JavaScript layer for the `state` field to get passed down? It feels like it might work out-of-the-box but if not then this might result in threads being dropped from JS payloads. Just something to check when updating RN.

Reply:
The ReactNative JS layer doesn't directly know about threads, however the typescript definitions in `common.d.ts` will need to be changed to reflect the `state` property.